api/model/src/main/java/org/projectnessie/api/v2/doc/ApiDoc.java

```diff
@@ -42,38 +42,18 @@ public interface ApiDoc {
           + "at different times depending on the state of the system. Using the full 'name@hash' form is recommended "
           + "to avoid ambiguity.\n";
 
-  String BRANCH_DESCRIPTION =
+  String COMMIT_BRANCH_DESCRIPTION =
       "A reference to a particular version of the contents tree (a point in history) on a branch.\n"
-          + "This reference can be specified in these forms:\n"
-          + "- \\- (literal minus character) - Identifies the HEAD of the default branch \n"
-          + "- name - Identifies the HEAD commit on the named branch\n"
+          + "This reference is specified in this form:\n"
```

Comment:
I think I opened a can of works :-) In fact, even if the REST API accepts references without hashes, the Service layer does not. IOW the only form of reference that works is the full `name@hash` form.

Reply:
Note that the client API allows the commit operation to be sent without hash:

```java
api()
    .commitMultipleOperations()
    .branch(Branch.of("main", null)) // null hash allowed here
    .operation(Put.of(a, ta))
    .commitMeta(CommitMeta.fromMessage("commit 1"))
    .commit();
```

This results in the following error:

```
Bad Request (HTTP/400): commitMultipleOperations.expectedHash: must not be null
org.projectnessie.error.NessieBadRequestException: Bad Request (HTTP/400): commitMultipleOperations.expectedHash: must not be null
	at org.projectnessie.error.ErrorCode.lambda$asException$1(ErrorCode.java:66)
	at java.base@11.0.18/java.util.Optional.map(Optional.java:265)
	at org.projectnessie.error.ErrorCode.asException(ErrorCode.java:66)
```

Reply:
IIRC older versions allowed to commit without an expected hash. Probably a relict from that time.

Reply:
Can of worms... perhaps :) I believe `-` (default branch) should also be supported... the original idea was to make it easier for hand-written `curl` commands.

Reply:
In fact, you can use the dash... if you also provide the hash, e.g. `-@cafebabe` :-) Which, I suppose, defeats the initial purpose of the dash. But unfortunately any references without explicit hashes are currently refused (and were already before this PR).

Reply:
For future reference, here is the status of `expectedHash` parameter across different layers and for different operations.

| Operation | REST layer (TreeResource) | Service layer (TreeService) | Version Store layer |
|--------|--------|--------|--------|
| Commit | Not required (path param), defaults to null | Required | Not required, defaults to HEAD |
| Merge | Required (request body) | Required | Not required, defaults to HEAD |
| Transplant | Required (request body) | Required | Not required, defaults to HEAD |